### PR TITLE
Allow users to set Latest Completed ToDos for all cars in new user setup

### LINKED ISSUE #194: Allow for setting the last todo completion date for all cars in setup
Currently the user can only set the last todo completion for the first car that was created, but there is no notice telling them which car the tasks apply to.

Ideally the user should be able to create past todos for all cars, but if not then there needs to be a note explaining the situation.

### PR DESCRIPTION
**Pull request by baylessj**
Closes #194.

### CODE REVIEW

**Comment by DavBfr** [comment]
What is the goal of this?

**Comment by baylessj** [comment]
When my parents were first setting up their accounts, they added two cars in the first of the new user setup screens. The next setup screen asks for the last time that you changed the car's oil and tires, but it didn't specify which of the two cars that it was for, and only offered 1 car's worth of input fields. This allows the user to set the latest oil change and tire change mileages for all of the car's that they added in the first new user setup screen.

I removed you from the review for now because I realized that I forgot to check if the tests were working. Turns out, unsurprisingly, they aren't working. I'm going to grab dinner now but I'll get the tests passing later this evening and add you back to the review.

**Comment by baylessj** [comment]
Added you back to review the changes, unit tests are back to working now.

I agree that the onboarding will likely get more complicated in the future, and that it will make sense to set up cars one by one. That will be a pretty big reworking and would tie well into our idea of capping free tier users at one car, so maybe that could be a task for sometime soon? I can start putting together some Figma sketches or something to brainstorm the screenflow for that when the time comes.

**Comment by DavBfr** [comment]
With the next PR, we'll have to add units validation first: you don't want to manually convert your mileage to kilometers if you're in the US ;)

File: lib/screens/welcome/views/new_user_setup/latestcompleted.dart [hunk 1]
```diff
@@ -10,44 +10,160 @@ import 'package:json_intl/json_intl.dart';
...
+  final bool showName;
+
+  @override
+  _TodoFieldsState createState() => _TodoFieldsState(c, formKey, showName);
+}
+
+class _TodoFieldsState extends State<_TodoFields> {
+  _TodoFieldsState(this.c, this.formKey, this.showName);
+
+  final Car c;
```
Comment (DavBfr) [inline lib/screens/welcome/views/new_user_setup/latestcompleted.dart]: you don't need to repeat the variables from the StatefulWidget. You can get the values using `widget.c`

**Pull request merged**